ui/org.eclipse.pde.launching/src/org/eclipse/pde/launching/AbstractPDELaunchConfiguration.java

```diff
@@ -189,6 +193,29 @@ private String[] updateVMArgumentWithAdditionalArguments(String[] args, boolean
 		return args;
 	}
 
+
+	private boolean isEclipseBundleGreaterThanVersion(int major, int minor) {
+		PDEState pdeState = TargetPlatformHelper.getPDEState();
+		if (pdeState != null) {
+			try {
+				List<IPluginModelBase> platformBaseModel = Arrays.asList(pdeState.getTargetModels()).stream().filter(x -> Objects.nonNull(x.getBundleDescription())).filter(x -> Objects.nonNull(x.getBundleDescription().getSymbolicName())).filter(x -> x.getBundleDescription().getSymbolicName().equals("org.eclipse.platform"))//$NON-NLS-1$
```

Comment:
I haven't checked if that all makes sense or not, just a style hint.

Two issue I personally have with long chained code that might be cool to write but no fun to read and to debug:
1) it is hard to read & debug
2) filtering for `Objects.nonNull(x.getBundleDescription().getSymbolicName()))` would be not needed if you switch the order of next comparison to `"org.eclipse.platform".equals()`
3) You probably don't need a list anyway, so you could use `findFirst()` at the end

Reply:
ok, we will still need one not null filter(x -> Objects.nonNull(x.getBundleDescription()) before running through x.getBundleDescription().getSymbolicName().
I also prefer to go the normal for each loop way for read and debug but this might be faster I guess and  if we provide normal for loop codes, we get suggestions to use new ways to attract contributions.